.github/workflows/NativeTests.yaml

```diff
@@ -89,7 +90,7 @@ jobs:
         id: intTest3
         continue-on-error: true
         run: |
-          EXCLUDED_MODULES=spring-cloud-spanner-spring-data-r2dbc MODULE_UNDER_TEST="${{ matrix.it }}" ./.github/workflows/scripts/native-image-tests.sh
+          EXCLUDED_MODULES=spring-cloud-spanner-spring-data-r2dbc,spring-cloud-gcp-samples/spring-cloud-gcp-integration-pubsub-sample-test MODULE_UNDER_TEST="${{ matrix.it }}" ./.github/workflows/scripts/native-image-tests.sh
```

Comment:
Calling out this exclusion for `spring-cloud-gcp-samples/spring-cloud-gcp-integration-pubsub-sample-test`. This module verifies the interaction between the SenderApplication in `spring-cloud-gcp-integration-pubsub-sample-sender` and ReceiverApplication in `spring-cloud-gcp-integration-pubsub-sample-receiver`, the code for which is already being tested by `SenderIntegrationTest` and `ReceiverIntegrationTest`.

 More on the issue itself: The [setup of `spring-cloud-gcp-samples/spring-cloud-gcp-integration-pubsub-sample-test`](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/4329dfaf5e6dbbef4441e36b5b9205b6b1b209a7/spring-cloud-gcp-samples/spring-cloud-gcp-integration-pubsub-sample/spring-cloud-gcp-integration-pubsub-sample-test/src/test/java/com/example/SampleAppIntegrationTest.java#L39) is somewhat deviating from the convention of using `@SpringBootTest` and results in:

```
 JUnit Jupiter:SampleAppIntegrationTest:testSample(CapturedOutput)
    MethodSource [className = 'com.example.SampleAppIntegrationTest', methodName = 'testSample', methodParameterTypes = 'org.springframework.boot.test.system.CapturedOutput']
    => java.lang.IllegalArgumentException: Could not find class [org.graalvm.junit.platform.NativeImageJUnitLauncher__ApplicationContextInitializer]
       org.springframework.util.ClassUtils.resolveClassName(ClassUtils.java:341)
       org.springframework.context.aot.AotApplicationContextInitializer.instantiateInitializer(AotApplicationContextInitializer.java:80)
       org.springframework.context.aot.AotApplicationContextInitializer.initialize(AotApplicationContextInitializer.java:71)
       org.springframework.context.aot.AotApplicationContextInitializer.lambda$forInitializerClasses$0(AotApplicationContextInitializer.java:61)
       org.springframework.boot.SpringApplication.applyInitializers(SpringApplication.java:610)
       [...]
     Caused by: java.lang.ClassNotFoundException: org.graalvm.junit.platform.NativeImageJUnitLauncher__ApplicationContextInitializer

```
As described in https://github.com/spring-projects/spring-boot/issues/35626, while usage of `SpringBootTest` is supported out-of-the-box, this particular scenario of using SpringApplicationBuilder seems to cause the aot processing step to not be triggered correctly.

Reply:
@burkedavison @zhumin8 wanted to get your thoughts on this? Are we okay with excluding this test (`SampleAppIntegrationTest`) since it is duplicating test coverage that is accounted for ReceiverIntegrationTest and SenderIntegrationTest? Or would it be worth exploring this further?

Reply:
As mentioned in the ticket, have we tried explicitly setting the main class to https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/main/spring-cloud-gcp-samples/spring-cloud-gcp-pubsub-sample/src/main/java/com/example/PubSubApplication.java ?

Reply:
Thank you for the tip! Unlike `spring-cloud-gcp-pubsub-sample`, this module didn't have an application class so I implemented a new application that replicates the behavior in SampleAppIntegrationTests and set it to the `main-class` config in spring-boot-maven-plugin ([d883491](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/pull/2007/commits/d883491fc93466aac9dac43c44a3165e8b8bd3f9) and [5588fb0](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/pull/2007/commits/5588fb04f19c3dad755a5d8033bc9b4a60747d40)). However, the error still persists.

Reply:
[Update] Including `WebEnvironment` information to the `@SpringBootTest` annotation ended up doing the trick!